public/app/features/variables/datasource/reducer.ts

```diff
@@ -68,9 +69,13 @@ export const dataSourceVariableSlice = createSlice({
 
       instanceState.options = options;
     },
+    changeMulti: (state: VariablesState, action: PayloadAction<VariablePayload<boolean>>) => {
+      const instanceState = getInstanceState<DataSourceVariableModel>(state, action.payload.id);
+      changeMultiTo(instanceState, action.payload.data);
+      return state;
+    },
```

Comment:
is this really needed in each type reducer? is the logic in the shared reducer not enough?

Reply:
I was split between adding it in the shared reducer or for the ones that actually support multi and ended with adding it to the variable types supporting multi to keep the shared reducer free from special cases.